Automattic/jetpack
# required-review: Don't fail on case mismatch for `@singleuser` pseudo-teams

## Proposed changes:

GitHub usernames are case-insensitive but case-preserving. To match that logic, we want `@singleuser` to match `SingleUser` or whatever the actual username is.

This is complicated by the fact that "case-insensitive" can vary depending on what exactly GitHub is doing on their end, what version of Unicode they're using, and so on. So let's just ask GitHub's API for the correct casing of the username and use that instead of trying to work out all those edge cases ourself.

### Other information:


## Jetpack product discussion


https://github.com/Automattic/jetpack/pull/29318#issuecomment-1456890325

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Try changing the case of an `@singleuser` entry in `.github/files/required-review.yaml` and using the `test.sh` script to see how it handles an appropriate PR.
